### DISCUSSION TITLE
Module dependency for common module

### DISCUSSION THREAD

**Discussion by shreyy1234** [spring-projects/spring-modulith#370] [Q&A]
I have 3 modules, in this one module name is common.
common
|- datatype
|- infrastructure
|- enums
|- Aggregates

common module has different packages. As this module in common its internal classes of internal packages should be easily accessible.
However using modulith, module. verify fails when I try to use an enum from this common module in any other module.
I can able to resolve this by adding package-info into each subpackage of common module e.g
@org.springframework.modulith.NamedInterface("datatype")
package com.common.datatype;

however, adding package-info into each package is not a clean solution. is there any other way where I could mention this is a shared module so the modularity test should not fail?
Already tried with but it did not work.
@Modulithic(sharedModules = "com.common")
public class Application {

**Comment by onjik** [comment]
I'm not sure because I haven't tested it, but @Modulthic::sharedModule seems to write the correct package name. {com.common.enum,com.comm.datatype…} like this. In the case of Java's package, it looks hierarchical to see, but I know it is actually recognized as a separate, independent place.
I haven't tested it, so try it.

**Comment by shreyy1234** [comment]
tried @Modulithic(sharedModules = "com.common", additionalPackages = "com.common.datatype", useFullyQualifiedModuleNames = true) but it does not work without packge-info. getting

org.springframework.modulith.core.Violations: - Module 'com.other' depends on non-exposed type com..common.datatype.testdata within module 'com.common'!

**Reply by peterkronenberg** [reply]
Did you ever find a solution to this?

**Reply by cjacky475** [reply]
@shreyy1234, @peterkronenberg, have you guys found out a solution or is this not yet available in this library?

**Comment by luciano-fiandesio** [comment]
Also interested…Is it possible in Modulith to "mark" a first-level package as "non module" (and propagate to all sub-packages)?

**Comment by emileastih1** [comment]
Hello,
Have you tried declaring your "common" package as OPEN in a package-info.java?

You can do so like this:

@org.springframework.modulith.ApplicationModule(
type = Type.OPEN
)
package example.common;

Reference: https://docs.spring.io/spring-modulith/reference/fundamentals.html

**Reply by shwetakawale** [reply]
@emileastih1  do we need to impot anything for "Type", its giving me compilation error, i have upgraded spring modulith to 1.2.1, still same issue

**Reply by luciano-fiandesio** [reply]
I had the same problem. Check out the Javadoc: https://docs.spring.io/spring-modulith/docs/1.2.1/api/org/springframework/modulith/ApplicationModule.html#type()

**Reply by emileastih1** [reply]
@shwetakawale yes you need to import the Type from the modulith.ApplicationModule package, here is an example of how the package-info file shoud look like:

```
@ApplicationModule(type = Type.OPEN)
package org.springsource.restbucks.legacy;

import org.springframework.modulith.ApplicationModule;
import org.springframework.modulith.ApplicationModule.Type;
```

Note that in my example I am using Java 22 and <spring-modulith.version>1.2.0</spring-modulith.version>.

**Comment by simasch** [comment]
I exclude such packages because IMO they shouldn't be modules:

```
 ApplicationModules modules = ApplicationModules.of(Application.class,
            JavaClass.Predicates.resideInAnyPackage("example.configuration..", "example.common..");
```

**Comment by AJ198081** [comment]
Thanks @simasch - facing similar issue of failing module test, and excluding packages doesn't seem to work either. Have created a new thread for the same issue, if someone else who experienced the similar issue can help out.
